gtnet/run.py

```diff
@@ -1,14 +1,48 @@
-from .model import load_model
-from .sequence import chunk_sequences
+from model import load_model
+from sequence import _get_DNA_map, get_sequences, get_bidir_seq
+import numpy as np
+import scipy.special
+import argparse
+import logging
 
-def predict(sequences=None, **kwargs):
-    if fasta_path is None:
-        raise ValueError('sequences must not be None')
 
-    model = load_model()
+# should this be in another file?
+def get_prob(x):
+    soft_out = scipy.special.softmax(x)
+    probabilities = soft_out.mean(axis=0)
+    return probabilities
+
+
+def predict(fasta_path, model_path, **kwargs):
+    if fasta_path is None:
+        logging.error('Please provide a fasta path!')
+        exit()
 
-    for seq in chunk_sequences(fasta_path):
+    model = load_model(model_path)
+    input_name = model.get_inputs()[0].name
+    chars, basemap = _get_DNA_map()
+    for seq in get_sequences(fasta_path, basemap):
         # 1. chunk sequences
+        bidir_seq = get_bidir_seq(seq)
         # 2. pass chunks into model
-        # 3. softmax outputs
-        # 4. return probabilities
+        output = model.run(None, {input_name: bidir_seq.astype(np.int64)})[0]
+        # 3. get probalities
+        probabilities = get_prob(output)
```

Comment:
This shouldn't need to be called, since I added softmax to the ONNX model

Reply:
removed